core/eth/contracts/FluxAggregator.go

```diff
@@ -0,0 +1,181 @@
+package contracts
+
+import (
+	"math/big"
+
+	"chainlink/core/eth"
+	"chainlink/core/store/models"
+	"chainlink/core/utils"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+)
+
+//go:generate mockery -name FluxAggregator -output ../../internal/mocks/ -case=underscore
+
+type FluxAggregator interface {
+	eth.ConnectedContract
+	SubscribeToLogs(fromBlock *big.Int) (LogSubscription, error)
+	LatestAnswer(precision int32) (decimal.Decimal, error)
+	LatestRound() (*big.Int, error)
+	ReportingRound() (*big.Int, error)
+	TimedOutStatus(round *big.Int) (bool, error)
+	LatestSubmission(oracleAddress common.Address) (*big.Int, *big.Int, error)
+}
+
+type fluxAggregator struct {
+	eth.ConnectedContract
+	ethClient eth.Client
+	address   common.Address
+}
+
+type LogNewRound struct {
+	RoundId   *big.Int
+	StartedBy common.Address
+	StartedAt *big.Int
+	Address   common.Address
+}
+
+type LogRoundDetailsUpdated struct {
+	PaymentAmount  *big.Int
+	MinAnswerCount *big.Int
+	MaxAnswerCount *big.Int
+	RestartDelay   *big.Int
+	Timeout        *big.Int
+	Address        common.Address
+}
+
+type LogAnswerUpdated struct {
+	Current   *big.Int
+	RoundID   *big.Int
+	Timestamp *big.Int
+	Address   common.Address
+}
+
+func NewFluxAggregator(ethClient eth.Client, address common.Address) (FluxAggregator, error) {
+	contract, err := eth.GetV6Contract(eth.FluxAggregatorName)
+	if err != nil {
+		return nil, err
+	}
+	connectedContract := eth.NewConnectedContract(contract, ethClient, address)
+	return &fluxAggregator{connectedContract, ethClient, address}, nil
+}
+
+func (fa *fluxAggregator) SubscribeToLogs(fromBlock *big.Int) (LogSubscription, error) {
```

Comment:
The integration tests in features_test.go exercise subscriptions + decoding the `NewRound` event, which is the only one we're actually using right now.

Note: this will change in https://github.com/smartcontractkit/chainlink/pull/2458 wherein I've put together a more general way of handling logs and subscriptions -- there are a bunch of very explicit tests around this functionality in that one.